anaconda_project/requirements_registry/providers/conda_env.py

```diff
@@ -169,7 +175,9 @@ def provide(self, requirement, context):
         env_name = context.status.analysis.config.get('env_name', context.default_env_spec_name)
         env_spec = requirement.env_specs.get(env_name)
 
-        if context.status.analysis.config['source'] == 'inherited':
+        if self._bootstrap:
+            prefix = os.path.join(project_dir, 'envs', 'bootstrap-env')
```

Comment:
This really is it: the only material difference between CondaEnvProvider and CondaBootstrapEnvProvider is a hardcoded environment name.